### ISSUE TITLE
Abort outstanding HTTP requests when subscription is closed

### ISSUE THREAD

**Issue by Moezalez** [GoogleCloudPlatform/gcloud-node#1104]
Hi!

After a certain event I just want my app to shut down. It seems, though, that gcloud (where I only use pubsub) keeps my App alive. Is there a way to kill pubsub so my App can shut down?

**Comment by stephenplusplus** [comment]
It is probably a subscription object that is still pulling for new messages. You can quickly stop that process by calling subscription.removeAllListeners()

**Comment by Moezalez** [comment]
Hey @stephenplusplus, thanks for your answer.
That did not work for me.
Here a little example of what I mean. This will stay alive:

```JavaScript
  let pubsub = gcloud.pubsub({
    projectId: projectId,
    keyFilename: keyPath
  });

  let update = pubsub.topic('someTopic');

  update.subscribe('update', {reuseExisting: true}, function (error, updateSub) {
    if (error) {
      return console.log(error);
    }

    updateSub.on('error', function (error) {
      console.log(error);
    });

    updateSub.on('message', function (message) {
      console.log(message);
    });

    setTimeout(function () {
      updateSub.removeAllListeners();
    }, 500);
  });
```

**Comment by stephenplusplus** [comment]
Got to the bottom of this! Unfortunately, there are a few pieces involved...

1) Removing listeners is the right way to do this :+1:, but our code doesn't abort the HTTP request that remains open while waiting for new messages. We have to fix that (pr incoming soon)
2) We need a way to abort requests, but we're limited by one of the libraries we use, `retry-request`. I just fixed that: https://github.com/stephenplusplus/retry-request/commit/ecb30d01251f14079e4530f32291c5f050af9285
3) We need a way for the underlying library, `request` to allow us to abort a request that has a `timeout` set. PR sent for that here: https://github.com/request/request/pull/2052

**Comment by stephenplusplus** [comment]
This is actually going to be resolved by #1070, since we're switching to using [`gRPC`](https://github.com/grpc/grpc). If all goes well, a release that has this completely resolved will be put out on Monday.
